src/contexts/providers.context.tsx

```diff
@@ -221,6 +221,12 @@ const ProvidersProvider: FC = (props) => {
     [connectedProvider]
   );
 
+  useEffect(() => {
+    if (account.status === "failed") {
```

Comment:
I think that the account can also has the status `failed` in the login page, wouldn't this trigger an infinite loop or a refresh?

Reply:
Not sure I follow... the account's state is only stored in the provider. What we do on the log-in page is reset the state of `selectedWallet` when the account is `undefined`.

With this addition, when `account.status === "failed"` we will do 2 things:
* In `login.view.tsx` we reset the `selectedWallet`, which is ok since the state of the selected wallet is managed there.
* In `providers.context.tsx` we redirect to the path `/login` so if we were not yet there, we go there and do nothing otherwise.

What could cause an infinite loop?

Reply:
> What happens if we are in the `login` page and we redirect again to the `login`?

In my initial implementation, I was checking the current location and only redirecting to login when we were not on the login page. Later I confirmed that the router does nothing when we redirect to login from login so I removed this condition. From my tests, it works ok.